### DISCUSSION TITLE
How to use ngx-ui-tour with rxjs 6.6.7 version.

### DISCUSSION THREAD

**Discussion by NivRash1251** [hakimio/ngx-ui-tour#146] [Q&A] [answered]
My angular app is in Angular 15 version and RXJS 6.6.7. But for ngx-ui-tour Angular 15 requires RXJS 7.  How to resolve this compatibility issue ?

**Comment by hakimio** [comment]
Upgrade your `rxjs` to `v7`. The upgrade is pretty simple since there are very few breaking changes in `v7`.

**Comment by hakimio** [comment]
You can find a list of breaking changes [here](https://rxjs.dev/deprecations/breaking-changes).

**Reply by NivRash1251** [reply]
Upgrade is not possible for now as this is an enterprise level application and upgrading giving error in prerender-stage step.  Error is mentioned below:

Prerendering 1 route(s) to filepath/browser...ERROR ReferenceError: window is not defined
    at new TourResizeObserverService (file/main.js)
    at Object.TourResizeObserverService_Factory [as factory] (file/main.js)
    at R3Injector.hydrate (file/main.js)
    at R3Injector.get (file/main.js)
    at injectInjectorOnly (file/main.js)
    at ɵɵinject (file/main.js)
    at inject file/main.js)
    at new TourBackdropService (file/main.js)
    at Object.TourBackdropService_Factory [as factory] file/main.js)
    at R3Injector.hydrate (file/main.js)
✖ Prerendering routes to file/browser failed.

**Reply by hakimio** [reply]
Then you'll have to use older library version for Angular 14, which supports rxjs v6, or create and maintain your own fork of the library. The older version for Angular 14 is also compatible with Angular 15, but you might have to "force" your npm dependencies.
Since it's not "enterprise level" library and I am maintaining it for free, I have no plans to do LTS releases.

**Reply by hakimio** [reply]
BTW, the SSR error is not related to rxjs and should be fixed in the latest version of the library.

**Reply by hakimio** [reply]
Just remembered: the older version for Angular 14 is compatible with Angular 15, but most likely will not work correctly with "Angular Material" v15 UI components.

**Reply by NivRash1251** [reply]
Thank you for your response. I tried using different versions of RXJS and ngx-ut-tour in my application and nothing seems to work. Could you please help to check if this library is SSR compatible ?

**Reply by hakimio** [reply, marked as answer]
@NivRash1251 I have released `v11.0.4` which fixes your SSR issue.
